.buildkite/commands/build-for-testing.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+echo "--- :rubygems: Setting up Gems"
+restore_cache "$(hash_file .ruby-version)-$(hash_file Gemfile.lock)"
+gem install bundler
+bundle install
+save_cache vendor/bundle "$(hash_file .ruby-version)-$(hash_file Gemfile.lock)"
+
+echo "--- :cocoapods: Setting up Pods"
+
+# Caching the specs repos and global pod cache can dramatically improve Pod times
+restore_cache "$BUILDKITE_PIPELINE_SLUG-specs-repos"
+restore_cache "$BUILDKITE_PIPELINE_SLUG-global-pod-cache"
```

Comment:
Not sure if you're aware, but the caching for the specs repo is not working (or maybe I'm missing something on how it's supposed to work).

I re-run the pipeline for the last commit on this branch. The [original run](https://buildkite.com/wordpress-mobile/wordpress-ios/builds/1608#a864a310-4713-4054-9698-1a5e72525786/334) too ~18 minutes in the Pods step, the [second](https://buildkite.com/wordpress-mobile/wordpress-ios/builds/1653#9cc6304a-acbe-48ce-a7f8-f1931c0124ed/335) a mere ~2 minutes:

![image](https://user-images.githubusercontent.com/1218433/126922984-e48c5877-228f-4644-9ccc-0d51d81b4ea7.png)
![image](https://user-images.githubusercontent.com/1218433/126922988-29da9c97-b8af-4bcb-84fc-334211bcbd9f.png)

Both returned `No cache entry found` when attempting to `restore_cache` for the spec.